bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -323,12 +324,9 @@ protected void resetStatusBasedBackoff() {
    * @return a {@link ExponentialRetryAlgorithm} object.
    */
   private ExponentialRetryAlgorithm createRetryAlgorithm(ApiClock clock) {
-    long timeoutMs = retryOptions.getMaxElapsedBackoffMillis();
+    Optional<Long> operationTimeoutMs = deadlineGenerator.getOperationTimeoutMs();
 
-    Deadline deadline = getOperationCallOptions().getDeadline();
-    if (deadline != null) {
-      timeoutMs = deadline.timeRemaining(TimeUnit.MILLISECONDS);
-    }
+    long timeoutMs = operationTimeoutMs.or((long) retryOptions.getMaxElapsedBackoffMillis());
```

Comment:
Can you add a todo to try to move this logic into your deadline generator impl?

Reply:
Added a TODO comment. I did consider that option (basically having the backoffmillis as a parameter to getOperationTimeoutMs). I didn't as it ended up just being a fallback option that Optional easily represented. I'm not strongly in favor either way though, so I guess a TODO is fine for now